accounts/tests.py

```diff
@@ -127,23 +156,156 @@ def test_account_rejection(self, mock_post, mock_get):
+        result = agol.add_to_group('username', 'group')
+        self.assertFalse(result)
+
+        mock_response.json.return_value = {"notAdded": []}
+        result = agol.add_to_group('username', 'group')
+        self.assertTrue(result)
+
+    @patch.dict('os.environ', {'search': 'Smo.Joe_EPAEXT'})
+    def test_get_requests_by_search_param(self):
+        if 'search' in os.environ:
+            search_text = os.environ['search']
+            search_params = Q()
+            for search_field in AccountViewSet.search_fields:
+                search_params.add(Q(**{search_field + '__contains': search_text}), Q.OR)
+            search_results = AccountRequests.objects.filter(search_params)
+            self.assertTrue(len(list(search_results)) == 1)
+
 
     def test_invitations(self):
         agol = AGOL.objects.get(pk=1)
         agol.get_token = MagicMock(return_value='token')
-        requests = AccountRequests.objects.all()
+        requests = AccountRequests.objects.filter(agol_id=None)
         invitations = agol.generate_invitations(requests, 'password')
-        self.assertTrue('000b6ee121bb4739a5021e0f0241ff01' in invitations[0]["groups"])
-        self.assertTrue('00237b674c3347a18e0fb2bfcdb248e1' in invitations[0]["groups"])
+        self.assertTrue(requests[0].email in invitations[0]["email"])
+        self.assertTrue(requests[0].first_name in invitations[0]["firstname"])
+        self.assertTrue(requests[0].last_name in invitations[0]["lastname"])
+
+    def test_user_has_account_request_permission(self):
+        request = self.factory.get('/account/approvals')
+
+        # does sponsor have permission
+        sponsor = User.objects.get(pk=1)
+
+        request.user = sponsor
+        for ac in AccountRequests.objects.all():
+            result = IsSponsor().has_object_permission(request, SponsorsViewSet, ac)
+            self.assertTrue(result)
+
+        # does unrelated user have permission
+        unrelated_user = User.objects.get(pk=2)
+        request.user = unrelated_user
+        for ac in AccountRequests.objects.all():
+            result = IsSponsor().has_object_permission(request, SponsorsViewSet, ac)
+            self.assertFalse(result)
+
+        # after making delegate does user have permission
+        sponsor.agol_info.delegates.set([unrelated_user])
+        for ac in AccountRequests.objects.all():
+            result = IsSponsor().has_object_permission(request, SponsorsViewSet, ac)
+            self.assertTrue(result)
+
+
+class TestAGOLGroups(TestCase):
+    fixtures = ['fixtures.json']
+
+    # Get AGOLGroups by param tests #
+    @patch.dict('os.environ', {'all': 'true'})
+    def test_get_agol_groups_by_all_param(self):
+        if 'all' in os.environ:
+            if os.environ.get('all', 'false') == 'true':
+                all_results = list(AGOLGroup.objects.all())
+                self.assertIsNotNone(all_results)
+
+    @patch.dict('os.environ', {'is_auth_group': 'true'})
```

Comment:
@tommckennon why are you patching os.environ.  If you just want to create a dict... just create it in the test.  Also not sure what you are testing... again it looks like testing django code.

Reply:
Removed